objects/src/transaction/inputs.rs

```diff
@@ -327,64 +326,95 @@ fn build_nullifier_commitment<T: ToNullifier>(notes: &[T]) -> Digest {
 // INPUT NOTE
 // ================================================================================================
 
+const AUTHENTICATED: u8 = 0;
+const UNVERIFIED: u8 = 1;
+
 /// An input note for a transaction.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct InputNote {
-    note: Note,
-    proof: NoteInclusionProof,
+pub enum InputNote {
+    /// Input notes which have an authentication path and are authenticated by the transaction
+    /// kernel.
+    Authenticated { note: Note, proof: NoteInclusionProof },
+
+    /// Input notes which do not yet an authentication path. These are consumed by the transaction
+    /// kernel, and later on authenticated by the rollup kernels.
+    Unverified { note: Note },
 }
 
 impl InputNote {
     /// Returns a new instance of an [InputNote] with the specified note and proof.
-    pub fn new(note: Note, proof: NoteInclusionProof) -> Self {
-        Self { note, proof }
+    pub fn new_authenticated(note: Note, proof: NoteInclusionProof) -> Self {
+        Self::Authenticated { note, proof }
     }
```

Comment:
Should we also add a constructor for the "unauthenticated" variant? Or maybe we don't need either of the constructors?

Reply:
added